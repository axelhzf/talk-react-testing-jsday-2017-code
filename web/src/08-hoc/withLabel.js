import React from 'react';
import './style.css';

export default function withLabel(Component) {
  return class WithLabel extends React.Component {
    render() {
      const { label, ...other } = this.props;
      return (
        <div className="withLabel">
          <label>{this.props.label}</label>
          <div className="content">
            <Component {...other} />
          </div>
        </div>
      );
    }
  };
}
