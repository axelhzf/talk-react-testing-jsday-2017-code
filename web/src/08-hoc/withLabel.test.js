import React from 'react';
import { shallow } from 'enzyme';
import withLabel from './withLabel';

describe('08-hoc', () => {
  it('should render the component', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <InputWithLabel value="Luke skywalker" onChange={onChange} label="Name" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

const Input = ({ value, onChange }) => (
  <input type="text" className="input" value={value} onChange={onChange} />
);
const InputWithLabel = withLabel(Input);
