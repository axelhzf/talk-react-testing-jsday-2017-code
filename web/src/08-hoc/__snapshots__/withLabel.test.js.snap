// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`08-hoc should render the component 1`] = `
<div
  className="withLabel"
>
  <label>
    Name
  </label>
  <div
    className="content"
  >
    <Input
      onChange={[Function]}
      value="Luke skywalker"
    />
  </div>
</div>
`;
